pkg/expression/constant.go

```diff
@@ -260,18 +248,15 @@ func (c *Constant) Eval(ctx sessionctx.Context, row chunk.Row) (types.Datum, err
 			return c.Value, nil
 		}
 		if c.DeferredExpr != nil {
-			if _, sfOk := c.DeferredExpr.(*ScalarFunction); sfOk {
```

Comment:
I think `DefferedExpr` should always be `ScalarFunction`, so this cast is meaningless now and I removed it. PTAL @qw4990